src/App.tsx

```diff
@@ -1,61 +1,133 @@
 import './App.scss';
+import React, { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [option, setOption] = useState(0);
+  const [title, setTitle] = useState('');
+  const [todos, setNewTodo] = useState(initialTodos);
+  const [titleError, setTitleError] = useState(false);
+  const [selectUserError, setSelectUserError] = useState(false);
+
+  useEffect(() => {
+    if (title === '') {
+      setTitleError(true);
+    } else {
+      setTitleError(false);
+    }
+  }, [title]);
+
+  useEffect(() => {
+    if (option === 0) {
+      setSelectUserError(true);
+    } else {
+      setSelectUserError(false);
+    }
+  }, [option]);
```

Comment:
let's show errors when user submits the form, not when he only opens the page

so we need to make this calculation in handleSubmit